### DISCUSSION TITLE
Using the vercel API to create a deploy hook

### DISCUSSION THREAD

**Discussion by marcusps11** [vercel/community#3383] [Help]
### Summary

I am trying to create a deploy hook programatically that i can then link to my project. I am using this api call

`(curl` -X POST https://api.vercel.com/v1/projects/<your_project_id>/deploy-hooks \
  -H "Authorization: Bearer <your_vercel_token>" \
  -H "Content-Type: application/json" \
  -d `'{"name":"my-hook","gitBranch":"main"}')`

However i get {"error":{"code":"not_found","message":"The project <your_project_id> was not found."}}%

I have triple checked the projectID and and the token is 100% right?  Am i using the wrong projectId?

### Steps to Reproduce

`(curl` -X POST https://api.vercel.com/v1/projects/<your_project_id>/deploy-hooks \
  -H "Authorization: Bearer <your_vercel_token>" \
  -H "Content-Type: application/json" \
  -d `'{"name":"my-hook","gitBranch":"main"}')`

**Comment by amyegan** [comment]
Hey @marcusps11. I'm not familiar with that endpoint. Where did you find it?

**Comment by marcusps11** [comment]
Is it not possible to create a deploy hook programmatically?  I got it off
chat gpt so it may not exist!

**Comment by amyegan** [comment]
Ah, I've seen a few of these slightly wrong recommendations from ChatGPT. 😅 

Since they don't need to be created very often for most workflows, it's usually easiest to [create new Deploy Hooks in the project settings UI](https://vercel.com/docs/concepts/deployments/deploy-hooks#creating-a-deploy-hook). There is not an official supported way to create Deploy Hooks with our public [Vercel REST API Endpoints](https://vercel.com/docs/rest-api/endpoints#). 

If you're using a CI/CD solution, you could [use the CLI in a custom workflow](https://vercel.com/guides/using-vercel-cli-for-custom-workflows) to deploy from any branch of your repo. That would let you avoid the need to create new Deploy Hooks.

There may be an unofficial way to do it if you _really_ need to. Just be aware that it is not officially supported and the endpoint could change or disappear at any time. If you take a look at the network log in dev tools when creating a new Deploy Hook, you'll see a call to a deploy-hooks endpoint. But if you can use the UI, I highly recommend sticking with that option
